tests/models/ensemble/bidaf_ensemble_test.py

```diff
@@ -0,0 +1,105 @@
+                {
+                        "span_start_probs": torch.autograd.Variable(torch.FloatTensor([[0.0009347021696157753, 0.0012768324231728911, 0.9977738261222839, 1.4630706573370844e-05]])),
+                        "span_end_probs": torch.autograd.Variable(torch.FloatTensor([[4.180105679552071e-05, 2.762122494459618e-05, 0.9158865809440613, 0.0840439721941948]])),
+                        "best_span": torch.autograd.Variable(torch.LongTensor([[2, 2]])),
+                        "best_span_str": "cheese",
+                        "question_tokens": ["What", "did", "Michael", "eat", "?"], "passage_tokens": ["Michael", "ate", "cheese", "."]
+                },
+                {
+                        "span_start_probs": torch.autograd.Variable(torch.FloatTensor([[0.9977738261222839, 0.0009347021696157753, 0.0012768324231728911, 1.4630706573370844e-05]])),
+                        "span_end_probs": torch.autograd.Variable(torch.FloatTensor([[0.9158865809440613, 4.180105679552071e-05, 2.762122494459618e-05, 0.0840439721941948]])),
+                        "best_span": torch.autograd.Variable(torch.LongTensor([[0, 0]])),
+                        "best_span_str": "What",
+                        "question_tokens": ["What", "did", "Michael", "eat", "?"], "passage_tokens": ["Michael", "ate", "cheese", "."]
+                }
+        ]
+
+        assert ensemble(0, subresults) == 0
+
+    def test_ensemble_chooses_highest_average_confidence(self):
+        subresults = [
+                {
+                        "span_start_probs": torch.autograd.Variable(torch.FloatTensor([[0.0009347021696157753, 0.0012768324231728911, 0.9977738261222839, 1.4630706573370844e-05]])),
+                        "span_end_probs": torch.autograd.Variable(torch.FloatTensor([[4.180105679552071e-05, 2.762122494459618e-05, 0.9158865809440613, 0.0840439721941948]])),
+                        "best_span": torch.autograd.Variable(torch.LongTensor([[2, 2]])),
+                        "best_span_str": "cheese",
+                        "question_tokens": ["What", "did", "Michael", "eat", "?"], "passage_tokens": ["Michael", "ate", "cheese", "."]
+                },
+                {
+                        "span_start_probs": torch.autograd.Variable(torch.FloatTensor([[0.9977738261222839, 0.0009347021696157753, 0.0012768324231728911, 1.4630706573370844e-05]])),
+                        "span_end_probs": torch.autograd.Variable(torch.FloatTensor([[0.9158865809440613, 4.180105679552071e-05, 2.762122494459618e-05, 0.0840439721941948]])),
+                        "best_span": torch.autograd.Variable(torch.LongTensor([[0, 0]])),
+                        "best_span_str": "What",
+                        "question_tokens": ["What", "did", "Michael", "eat", "?"], "passage_tokens": ["Michael", "ate", "cheese", "."]
+                }
+        ]
+
+        assert ensemble(0, subresults) == 0
+
+    def test_ensemble_chooses_highest_average_confidence_reverse(self):
```

Comment:
what's this test? just the same as the above with the examples in the reverse order?

Reply:
yes--I could delete the proceeding one.  I didn't want every test to be satisfiable with the answer 0.

Reply:
Yeah, just having one of these two is probably enough here.